src/overlay/Peer.cpp

```diff
@@ -1868,7 +2016,159 @@ Peer::PeerMetrics::PeerMetrics(VirtualClock::time_point connectedTime)
     , mDuplicateFloodMessageRecv(0)
     , mUniqueFetchMessageRecv(0)
     , mDuplicateFetchMessageRecv(0)
+    , mTxHashReceived(0)
     , mConnectedTime(connectedTime)
+    , mMessagesFulfilled(0)
+    , mBannedMessageUnfulfilled(0)
+    , mUnknownMessageUnfulfilled(0)
+{
+}
+
+bool
+Peer::isPullModeEnabled() const
+{
+    return mPullModeEnabled;
+}
+
+void
+Peer::queueTxHashToAdvertise(Hash const& txHash)
 {
+    mTxHashesToAdvertise.emplace_back(txHash);
```

Comment:
same thing here: as long as FLOOD_MAX_ADVERT_SIZE is less than or equal to the XDR limit, it's fine. But that limit should be enforced during the calculation of FLOOD_MAX_ADVERT_SIZE.

Reply:
I check that the size is below the xdr limit & the calculated max advert size is asserted to be less than or equal to xdr's max advert size.

Reply:
This is a footgun: there are several tx broadcast path (at least loadgen and txqueue) and I don't see how you can guarantee that `broadcastMessage` will not be called too many times.
The assert that you added is redundant with the internal checks from `mTxHashesToAdvertise` and crashes the app (which is what I was trying to avoid with this comment) - we should just drop txs on the floor in this case.

Reply:
Removed the assert and we will drop the txn in this case.